numba/targets/arraymath.py

```diff
@@ -3666,6 +3666,48 @@ def _is_nonelike(ty):
     return (ty is None) or isinstance(ty, types.NoneType)
 
 
+@overload(np.array)
+def np_array(a, dtype=None, copy=True):
+
+    # developer note... keep this function (type_can_asarray) in sync with the
+    # accepted types implementations below!
+    if not type_can_asarray(a):
+        return None
+
```

Comment:
I think that `dtype` and `copy` should be validated as being something acceptable type wise. Else this sort of thing will end up as multiple screens of traceback/exceptions:
```python
from numba import njit
import numpy as np

@njit
def foo():
    return np.array((1,2,3), dtype='float32') # oops, strings not permitted

x = foo()
print(x)
print(x.dtype)
```